include/lapack/lahqr.hpp

```diff
@@ -0,0 +1,419 @@
+/// @file lahqr.hpp
+/// @author Thijs, KU Leuven, Belgium
+/// Adapted from @see https://github.com/Reference-LAPACK/lapack/tree/master/SRC/dlahqr.f
+//
+// Copyright (c) 2013-2022, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef __LAHQR_HH__
+#define __LAHQR_HH__
+
+#include <complex>
+#include <iostream>
+#include <vector>
+
+#include "lapack/utils.hpp"
+#include "lapack/types.hpp"
+#include "lapack/larfg.hpp"
+#include "lapack/larf.hpp"
+#include "lapack/lahqr_eig22.hpp"
+#include "lapack/lahqr_shiftcolumn.hpp"
+#include "lapack/lahqr_schur22.hpp"
+
+namespace lapack
+{
+
+    /** lahqr computes the eigenvalues and optionally the Schur
+     *  factorization of an upper Hessenberg matrix, using the double-shift
+     *  implicit QR algorithm.
+     *
+     * @return  0 if success
+     * @return -i if the ith argument is invalid
+     * @return  i if the QR algorithm failed to compute all the eigenvalues
+     *            in a total of 30 iterations per eigenvalue. elements
+     *            i:ihi of w contain those eigenvalues which have been
+     *            successfully computed.
+     *
+     * @param[in] want_t bool.
+     *      If true, the full Schur factor T will be computed.
+     * @param[in] want_z bool.
+     *      If true, the Schur vectors Z will be computed.
+     * @param[in] ilo    integer.
+     *      Either ilo=0 or A(ilo,ilo-1) = 0.
+     * @param[in] ihi    integer.
+     *      The matrix A is assumed to be already quasi-triangular in rows and
```

Comment:
What is the definition of a quasi-triangular matrix? Maybe we should point out to a reference.

Reply:
A quasi-triangular matrix is upper triangular, but can have 2x2 blocks on the diagonal for imaginary eigenvalues in real matrices.
I guess it would be useful to have a specific description of what we mean by standard form.